Kobolds + Dextrous simple animal tweaks

Xhuis opened the pull request:
Work-in-progress, please do not merge.

:cl: Xhuis
add: A strange new species of creature has appeared in lavaland. They appear sentient, intelligent, and organized, but they are weak and typically don't stray far from home. These "kobolds" seem to be born of, and devoutly protect, an ash drake, dubbed "Mother Drake". Remain cautious while mining or exploring, as these kobolds are cunning trapmakers.
/:cl:

Fixes:
- Fixed some dextrous simple mob issues.
- Changed throw code - any mob can now throw items, although only carbons are able to by default.
- Added the Ashen language (`ASHEN` in the code). To anyone who doesn't understand it, Ashen sounds like hissing and whistling.
- The generic dextrous HUD now has an icon for throwing and catching.

Main changes:
- Added kobolds. Kobolds are slow and frail simple mobs (speed of 1 and 50 HP respectively) with dexterity, the ability to throw, and the Ashen language.
  - Kobolds spawn in a cavern with a prideful ash drake that serves as their nest. The drake will not attack kobolds and has a greatly shortened aggro radius for all mobs.
    - All other mining mobs and other ash drakes will still attack kobolds.
  - Although weak, kobolds are cunning trap artists and can create and utilize a variety of traps such as the goliath snare, which serves as a powerful leg-hold trap  that tremendously damages any goliath foolish enough to dig its tentacles underneath it. These traps can be utilized by anyone but only kobolds know how to create and perform maintenance on them.
  - The prideful ash drake is known as Mother Drake, and is the mother of the kobolds, who protect her with their lives. Mother Drake lays kobold eggs for every two dead or unconscious creatures she consumes. Consumption is indiscriminate; any valid creatures nest to Mother Drake can be devoured.
  - When Mother Drake dies, she will drop an anomaly core and cause all kobolds to lose their intelligence. They will no longer know any languages, lose their dexterity, and forget their own names.
  - When finished, kobolds will have a 50% chance to appear in place of an ash walker nest. Kobolds and ash walkers cannot exist simultaneously.

Commit:
Kobold work

kilog commented:
KOBOLDS

coiax commented:
Again, why are we trying to ignore the large amount of reagents and existing work we've already done for humanoids

Xhuis commented:
http://tuckerskobolds.com

@coiax What do you mean?

MMMiracles commented:
What if, like, instead of recreating the examine system specifically for kobolds, you placed it at the base simple_animal like the dexterous changes?

Xhuis commented:
Kobolds just have a unique override for `examine()`. It's done with tons of other mobs, too. I can probably see about it, though.

Man-Shroom- commented:
these should outright replace ash walkers

Supermichael777 commented:
ash walkers are kobolds alredy. anoying assholes

AlexCrimson commented:
Kobolds? That cute. Do they hunt for shinies?

If you intend to steal ideas from Dwarf Fortress then be sure to give them Blowguns and kewl ghetto poisons.

Niknakflak commented:
Kobolds aside, the dexterous changes are a huge :+1:

InsaneHyena commented:
When I memed for kobolds, I just wanted short ash walkers, but this is even better.

KorPhaeron commented:
Please put them in their own ruin so they can easily be enabled/disabled.

I'm a bit concerned that the way for miners to stop ash walkers 2.0 is having to fight a dragon

KorPhaeron commented:
I guess playing as trappers might be cool, I'd have to see. It means no surgery/chemistry/machines/armour crafting though. Making the murder role even more simplistic might make things worse.

Zilenan91 commented:
Kobolds should probably only need one corpse to procreate. Watchers will kill them pretty much immediately without them being able to do anything, as will most other things.

Armhulen commented:
WHO WAS THE ONE WHO SHILLED FOR KOBOLDS?

WAKA WAKA WAKA MOTHERFUCKERS :+1:

Commit:
Kobold work and nest separation

Kromgar commented:
Replace ashliggers with these guys

Xhuis commented:
I'm getting that a lot, but I'd like to have some playtesting before I do anything major.

Armhulen commented:
Where's my 'New in this Commit'

Xhuis commented:
There's only been one commit.

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mining_mobs.dm:
```diff
@@ -12,7 +12,7 @@
 	a_intent = "harm"
 	var/throw_message = "bounces off of"
-	var/icon_aggro = null // for swapping to when we get aggressive
+	var/icon_aggro = null // for swapping to when we get aggressived
```
...

Xhuis replied:
God damn it, I thought I got rid of that typo. Sorry...

RemieRichards commented on code/modules/mob/living/simple_animal/hostile/mining_mobs.dm:
```diff
@@ -501,13 +501,18 @@
+/obj/effect/goliath_tentacle/New(turf/T, mob/living/O)
+	..()
+	if(!O)
+		qdel(src)
```
rip spawning these in for events I guess.

Xhuis replied:
Eh, I'll cut it out.

Xhuis commented:
Holy crap, that rainbow of labels. Anyway, forgot to push this commit, so here we go.
* Separated kobold dens from ash walker dens. They can now spawn independently.
* Added trapper's bags that can hold kobold devices. They start with three goliath snares and two sinew nets.
* Goliath snares now start contracted and must be reset.
* Goliath snares now become translucent when placed.
* Kobolds are now vulnerable to stuns and can process reagents. In addition, they can see in the dark and are immune to ash storms.
* Added sinew nets. They can be thrown at mining creatures to immobilize them for 7.5 seconds. During this time, they cannot attack, although they can still use ranged attacks, i.e. watcher beams and legion skulls.

coiax commented:
I am still ambivalent about making kobolts non-humanoids.

BUT

I like the idea of them, and you shouldn't lose hope. Remember that free golems and ash walkers were introduced with far more minimal features. The basic strokes are all that you need to start out with, and the ~~players will guide us the rest of the way~~ players will allow us to know what to nerf.

Armhulen commented:
Give them blowdarts to knock out the humams

WJohn commented:
I think this PR needs a few more tags.

Armhulen commented:
> I think this PR needs a few more tags.

Xhuis commented:
Can we get a FUCK tag please?

Xhuis commented:
This is merge-ready once I fix conflicts and have sprites. Right now I have some really gimped dragonlike sprites that I can use but I'd rather have proper ones.

Imagebard commented:
Why does this work by replacing ash lizards 50% of the time instead of having its own roll?
It lowers the chance of having kobolds. Did you increase the weight of ash drakes, at least?

coiax commented:
> Separated kobold dens from ash walker dens. They can now spawn independently.

Imagebard commented:
fuck

optimumtact commented:
Stale

Xhuis commented:
OH GOD ANOTHER TAG AAAAAA
Real talk though, yeah. Not sure if I'm going to continue this

KorPhaeron commented:
not sure how trapper gameplay would work but I was curious to try it

Xhuis commented:
Playtesting isn't very interesting because of how the traps work. Right now it's just "kite goliath and drop a snare when it uses its tentacles", plus the fact that monsters in an area tend to stick around, so nothing could wander into your traps.

KorPhaeron commented:
complex ai hunting and migration when

Zilenan91 commented:
The traps seem like they would work better if they were on station against things that could actually intelligently move around. If you want to playtest try dropping a few kobolds on station in maint or something and see how they do.

AlexCrimson commented:
Needs more stuff yo.

Add some kind of monster drop from Watchers(Maybe an eye or something) that can be used to make some sort of freezing trap, like cryosting. Not sure if mobs a vulnerable to that. Having a 3rd trap requiring Legions drops would be neat, but i have no idea what it would do.

Some kind of ranged weapon. Blowgun would be the obvious choice as a said before. Poisons would be harder to add, i guess. You would need to add some kind of critter or fungus/plant for Kobolds to harvest and refine into various poisons, and some way to add the poison to darts.

Kobolds in Dwarf Fortress are shamanistic, so you could add Totems for them to build. Either as just a visual thing or for something like a health regen effect when near one. 

They also tend to like shiny things, so i think it would be neat if that was the motivation for their interaction with Humans. Rather than Ash Walkers and the whole "get off my land" lore, Kobolds are more interested in stealing gems/metals/shiny things from the Humans. Seeing as they have a dragon to worship, gathering treasure for it would be kewl.

Seeing as you are going for sneaky sneak Kobolds, give them the Chameleon super power that lets them fade away whilst standing still. Either as a standard thing or as a reward for gathering shiny things and giving them to the dragon.

Zilenan91 commented:
The chameleon power sounds like it would be a good addition to kobolds. Currently an assistant with a spear seems like they'll destroy them.

Armhulen commented:
Potion crafting, holy shit yes. It'd also liven up Lavaland, with mushrooms and stuff

Iamgoofball commented:
i can make a potion crafting system for reagents

optimumtact commented:
Ping me if you want to reopen, or just start a new pr

Xhuis commented:
Thanks
